### PR TITLE
TMDM-12752 [REST API] Get entity details

### PR DESCRIPTION
**Pull request by htyin**
https://jira.talendforge.org/browse/TMDM-12752
**What is the current behavior?** (You should also link to an open issue here)

**What is the new behavior?**

New API return data model by entity name

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: org.talend.mdm.core/src/com/amalto/core/storage/services/SystemModels.java [hunk 1]
```diff
@@ -100,6 +105,27 @@ private static void reloadDataModel(String modelName) {
         session.end();
     }
 
+    @GET
+    @Path("/entities/{entityName}")
+    @ApiOperation("Get the details of an entity according his identifier.")
```
Comment (lpw82gf) [inline org.talend.mdm.core/src/com/amalto/core/storage/services/SystemModels.java]: `Get the details of an entity according its identifier.`

File: org.talend.mdm.core/src/com/amalto/core/storage/services/SystemModels.java [hunk 2]
```diff
@@ -100,6 +105,27 @@ private static void reloadDataModel(String modelName) {
...
+    public Response getEntityDetails(@ApiParam("Entity name") @PathParam("entityName") String entityName) {
+        try {
+            MetadataRepositoryAdmin metadataRepositoryAdmin = ServerContext.INSTANCE.get().getMetadataRepositoryAdmin();
+            String dataModelId = DataModelUtil.getDataModelNameByEntityName(metadataRepositoryAdmin,
+                    DataModelUtil.getDataModelNames(), entityName);
+            if (StringUtils.isNotEmpty(dataModelId)) {
+                Map<String, Object> map = new HashMap<String, Object>();
```
Comment (lpw82gf) [inline org.talend.mdm.core/src/com/amalto/core/storage/services/SystemModels.java]: `Map<String, Object> map = new HashMap<>();`

File: org.talend.mdm.core/src/com/amalto/core/storage/services/SystemModels.java [hunk 3]
```diff
@@ -100,6 +105,27 @@ private static void reloadDataModel(String modelName) {
...
+                map.put("entity", entityName); //$NON-NLS-1$
+                map.put("dataModelId", dataModelId); //$NON-NLS-1$
+                return Response.ok(map).type(MediaType.APPLICATION_JSON_TYPE).build();
+            } else {
+                return Response.status(Response.Status.NOT_FOUND).build();
+            }
+        } catch (Exception e) {
+            throw new RuntimeException("Could not get data model.", e); //$NON-NLS-1$
```
Comment (lpw82gf) [inline org.talend.mdm.core/src/com/amalto/core/storage/services/SystemModels.java]:
```
LOGGER.error(responseMessage, e);
return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(responseMessage).build();
```
Deal with the exception as something like above.

File: org.talend.mdm.core/src/com/amalto/core/storage/services/SystemModels.java [hunk 4]
```diff
@@ -100,6 +105,27 @@ private static void reloadDataModel(String modelName) {
...
+            MetadataRepositoryAdmin metadataRepositoryAdmin = ServerContext.INSTANCE.get().getMetadataRepositoryAdmin();
+            String dataModelId = DataModelUtil.getDataModelNameByEntityName(metadataRepositoryAdmin,
+                    DataModelUtil.getDataModelNames(), entityName);
+            if (StringUtils.isNotEmpty(dataModelId)) {
+                Map<String, Object> map = new HashMap<>();
+                map.put("entity", entityName); //$NON-NLS-1$
+                map.put("dataModelId", dataModelId); //$NON-NLS-1$
+                return Response.ok(map).type(MediaType.APPLICATION_JSON_TYPE).build();
+            } else {
+                return Response.status(Response.Status.NOT_FOUND).build();
```
Comment (lpw82gf) [inline org.talend.mdm.core/src/com/amalto/core/storage/services/SystemModels.java]: return Response.status(Response.Status.NOT_FOUND).entity("Entity of " + entity + " doesn't exist.").build();

**Comment by htyin** [comment]
http://192.168.31.210:8080/view/Master/job/Master_03_TMDMEE_FOR-JIRA/1727/

File: org.talend.mdm.core/src/com/amalto/core/storage/services/SystemModels.java [hunk 5]
```diff
@@ -100,6 +105,30 @@ private static void reloadDataModel(String modelName) {
...
+    @GET
+    @Path("/entities/{entityName}")
+    @ApiOperation("Get the details of an entity according its identifier.")
+    public Response getEntityDetails(@ApiParam("Entity name") @PathParam("entityName") String entityName) {
+        try {
+            MetadataRepositoryAdmin metadataRepositoryAdmin = ServerContext.INSTANCE.get().getMetadataRepositoryAdmin();
+            String dataModelId = DataModelUtil.getDataModelNameByEntityName(metadataRepositoryAdmin,
+                    DataModelUtil.getDataModelNames(), entityName);
+            if (StringUtils.isNotEmpty(dataModelId)) {
+                Map<String, Object> map = new HashMap<>();
```
Comment (VictorLi-talendbj) [inline org.talend.mdm.core/src/com/amalto/core/storage/services/SystemModels.java]: Could we change Map<String, Object> to Map<String, String>?

**Pull request merged**